### PR TITLE
[cleanup] move constants to feature-specific files

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Move feature-specific constants from api/common to be near where they are used 

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Ensure the change is noop, i.e. the binary builds and runs without issue

### Checklist

- [X] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [X] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: api/datadoghq/v2alpha1/datadogagent_default.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ const (
 	defaultLogContainerLogsPath          string = "/var/lib/docker/containers"
 	defaultLogPodLogsPath                string = "/var/log/pods"
 	defaultLogContainerSymlinksPath      string = "/var/log/containers"
-	defaultLogTempStoragePath            string = "/var/lib/datadog-agent/logs"
+	DefaultLogTempStoragePath            string = "/var/lib/datadog-agent/logs"
```
Comment (khewonc) [inline api/datadoghq/v2alpha1/datadogagent_default.go:27]: Should this go in `api/datadoghq/common/const.go`?

File: internal/controller/datadogagent/feature/apm/const.go [hunk 2]
```diff
@@ -13,6 +13,10 @@ import (
 
 const (
 	apmRBACPrefix = "apm"
+
+	apmHostPortName          = "traceport"
+	apmSocketVolumeName      = "apmsocket"
+	apmSocketVolumeLocalPath = "/var/run/datadog"
```
Comment (khewonc) [inline internal/controller/datadogagent/feature/apm/const.go:19]: Should this be combined with dsd like we do with the hostpath `DogstatsdAPMSocketHostPath`? Or would it be better to separate `DogstatsdAPMSocketHostPath` into dsd and apm specific paths?

File: api/datadoghq/common/const.go [hunk 3]
```diff
@@ -124,56 +83,37 @@ const (
...
+	CertificatesVolumeName    = "certificates"
+	CertificatesVolumePath    = "/etc/datadog-agent/certificates"
+	AuthVolumeName            = "datadog-agent-auth"
+	AuthVolumePath            = "/etc/datadog-agent/auth"
+	InstallInfoVolumeName     = "installinfo"
+	InstallInfoVolumeSubPath  = "install_info"
+	InstallInfoVolumePath     = "/etc/datadog-agent/install_info"
+	InstallInfoVolumeReadOnly = true
+
 	DogstatsdHostPortName      = "dogstatsdport"
```
[the remark this answers is not in the document]
Comment (celenechang) [reply api/datadoghq/common/const.go:99]: yea, i intentionally punted on the ones that require some thought, for now..

File: internal/controller/datadogagent/feature/apm/const.go (hunk 2 above)
Comment (celenechang) [reply internal/controller/datadogagent/feature/apm/const.go:19]: i'm not sure, but i guess we can decide and stick with one approach. it certainly seems there is some deduplication we can do for this particular constant

**Review by khewonc — APPROVED** [review]

**Pull request merged**